### PR TITLE
[feature](windows-func) support windows function cume_dist

### LINKED ISSUE #30726: [Doris-Future] (enhance) Support more window function
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

We should add more window function to cover other similar products' implementation. Now it includes but not limited to:
VARIANCE --- overall variance
VAR_SAMP --- sample variance
STD --- overall standard deviation
STDDEV_SAMP --- sample std
COVAR_POP --- overall covariance
COVAR_SAMP --- sample cov
CORR --- correlation coefficient
percent_rank --- percentage ranking
cume_dist --- cumulative distribution

This issue could be handled by many people. If you'd like to participate in this issue, please contact me via my we-chat: torch-wood_

### Solution

The idea solution is, for each one participating in this, you could implement one window function.
The basic usage could be found at https://doris.apache.org/docs/dev/sql-manual/sql-functions/window-functions/window-function/
And the code could be found in `aggregate_function_window.h` and `aggregate_function_window.cpp`
Have a good tour coding!

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by nanfeng1999**
## Proposed changes

Issue Number: close #30726 

support windows function cume_dist

```
+--------------+------------+--------------+-----------------------------+
| sales_person | sales_date | sales_amount | cumulative_sales_percentage |
+--------------+------------+--------------+-----------------------------+
| Bob          | 2024-02-01 |         1500 |                         0.5 |
| Alice        | 2024-02-01 |         2000 |                           1 |
| Bob          | 2024-02-02 |         1200 |                         0.5 |
| Alice        | 2024-02-02 |         1800 |                           1 |
| Bob          | 2024-02-03 |         1900 |                        0.25 |
| Tom          | 2024-02-03 |         2000 |                        0.75 |
| Jerry        | 2024-02-03 |         2000 |                        0.75 |
| Alice        | 2024-02-03 |         2200 |                           1 |
+--------------+------------+--------------+-----------------------------+
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/aggregate_functions/aggregate_function_window.h [hunk 1]
```diff
@@ -231,6 +231,61 @@ class WindowFunctionPercentRank final
     void deserialize(AggregateDataPtr place, BufferReadable& buf, Arena*) const override {}
 };
 
+struct CumeDistData {
+    float64_t numerator = 0;
```
Comment (zclllyybb) [inline be/src/vec/aggregate_functions/aggregate_function_window.h:235]: It seems that you should include the particular header file to introduce `float64_t` for all architecture

**Comment by nanfeng1999** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM, @HappenLee @zhangstar333 PTAL

**Comment by zclllyybb** [comment]
run p0 10

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**